src/main/resources/templates/register.html

```diff
@@ -10,6 +10,7 @@
 <div class="container">
     <h2>Register</h2>
     <form action="/register" method="post">
```

Comment:
In oauth2Login(), you should check that the /login page correctly handles authorization and redirection errors after a successful login.